src/carplay.cs

```diff
@@ -1008,4 +1289,556 @@ interface CPTemplateApplicationDashboardScene {
+
+		[Export ("updateTemplates:")]
+		void UpdateTemplates (CPTemplate[] newTemplates);
+	}
+
+	interface ICPNowPlayingTemplateObserver { }
+
+	[NoWatch, NoTV, NoMac, iOS (14,0)]
+	[Protocol]
+	[Model]
+	[BaseType (typeof (NSObject))]
+	interface CPNowPlayingTemplateObserver
+	{
+		[Export ("nowPlayingTemplateUpNextButtonTapped:")]
+		void UpNextButtonTapped (CPNowPlayingTemplate nowPlayingTemplate);
+
+		[Export ("nowPlayingTemplateAlbumArtistButtonTapped:")]
+		void AlbumArtistButtonTapped (CPNowPlayingTemplate nowPlayingTemplate);
+	}
+
+	[NoWatch, NoTV, NoMac, iOS (14,0)]
+	[BaseType (typeof (CPTemplate))]
+	[DisableDefaultCtor]
+	interface CPNowPlayingTemplate
+	{
+		[Static]
+		[Export ("sharedTemplate", ArgumentSemantic.Strong)]
+		CPNowPlayingTemplate SharedTemplate { get; /*set;*/ }
+
+		[Export ("addObserver:")]
+		void AddObserver (ICPNowPlayingTemplateObserver observer);
+
+		[Export ("removeObserver:")]
+		void RemoveObserver (ICPNowPlayingTemplateObserver observer);
+
+		[Export ("nowPlayingButtons", ArgumentSemantic.Strong)]
+		CPNowPlayingButton[] NowPlayingButtons { get; }
+
+		[Export ("upNextButtonEnabled")]
+		bool IsUpNextButtonEnabled { [Bind ("isUpNextButtonEnabled")] get; set; }
+
+		[Export ("upNextTitle")]
+		string UpNextTitle { get; set; }
+
+		[Export ("albumArtistButtonEnabled")]
+		bool IsAlbumArtistButtonEnabled { [Bind ("isAlbumArtistButtonEnabled")] get; set; }
+
+		[Export ("updateNowPlayingButtons:")]
+		void UpdateNowPlayingButtons (CPNowPlayingButton[] nowPlayingButtons);
+	}
+
+	[NoWatch, NoTV, NoMac, iOS (14,0)]
+	[BaseType (typeof (CPButton))]
+	interface CPContactMessageButton
+	{
+		[Export ("initWithImage:handler:")]
+		[DesignatedInitializer]
+		IntPtr Constructor (UIImage image, [NullAllowed] Action<CPButton> handler);
+
+		[Export ("initWithPhoneOrEmail:")]
+		IntPtr Constructor (string phoneOrEmail);
+
+		[Export ("phoneOrEmail")]
+		string PhoneOrEmail { get; }
+	}
+
+	[NoWatch, NoTV, NoMac, iOS (14,0)]
+	[BaseType (typeof (NSObject))]
+	interface CPEntity : NSSecureCoding {}
+
+	[NoWatch, NoTV, NoMac, iOS (14,0)]
+	[BaseType (typeof (CPNowPlayingButton))]
+	interface CPNowPlayingShuffleButton {}
+
+	[NoWatch, NoTV, NoMac, iOS (14,0)]
+	[BaseType (typeof (CPNowPlayingButton))]
+	interface CPNowPlayingAddToLibraryButton {}
+
+	[NoWatch, NoTV, NoMac, iOS (14,0)]
+	[BaseType (typeof (CPNowPlayingButton))]
+	interface CPNowPlayingMoreButton {}
+
+	[NoWatch, NoTV, NoMac, iOS (14,0)]
+	[BaseType (typeof (CPNowPlayingButton))]
+	interface CPNowPlayingPlaybackRateButton {}
+
+	[NoWatch, NoTV, NoMac, iOS (14,0)]
+	[BaseType (typeof (CPNowPlayingButton))]
+	interface CPNowPlayingRepeatButton {}
+
+	[NoWatch, NoTV, NoMac, iOS (14,0)]
+	[Protocol]
+	interface CPListTemplateItem
+	{
+		[Abstract]
+		[NullAllowed, Export ("text")]
+		string Text { get; }
+
+		[Abstract]
+		[NullAllowed, Export ("userInfo", ArgumentSemantic.Strong)]
+		NSObject UserInfo { get; set; }
+	}
+
+
+#if false
```

Comment:
On purpose?

Reply:
Yes, see the linked issue: https://github.com/xamarin/maccore/issues/2298

Is it better to leave out for now until we know Apple is actually adding it?

Reply:
either way is fine
you can add a link to the issue in the source (since the original commit message takes longer to get)